arquillian-webcam-video-recorder/src/main/java/org/arquillian/extension/recorder/video/webcam/impl/WebcamRecorder.java

```diff
@@ -0,0 +1,177 @@
+ * limitations under the License.
+ */
+package org.arquillian.extension.recorder.video.webcam.impl;
+
+import com.github.sarxos.webcam.Webcam;
+import org.arquillian.extension.recorder.video.Video;
+import org.arquillian.extension.recorder.video.VideoConfiguration;
+import org.arquillian.extension.recorder.video.VideoType;
+import org.arquillian.extension.recorder.video.webcam.configuration.WebcamVideoConfiguration;
+import org.jboss.arquillian.core.spi.Validate;
+import org.jcodec.api.awt.SequenceEncoder;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ *
+ * @author <a href="mailto:asotobu@gmail.com">Alex Soto</a>
+ */
+class WebcamRecorder {
+
+    private static final Logger logger = Logger.getLogger(WebcamRecorder.class.getName());
+
+    public static final int DEFAULT_FRAMERATE = 20;
+
+    public static final int MIN_FRAMERATE = 0;
+
+    public static final int MAX_FRAMERATE = 100;
+
+    private int frameRate = DEFAULT_FRAMERATE;
+
+    private final Dimension screenBounds;
+
+    private volatile boolean running = false;
+
+    private Thread thread;
+
+    private File recordedVideo = null;
+
+    private Timer timer;
+
+    private VideoConfiguration configuration;
+
+    public WebcamRecorder(WebcamVideoConfiguration configuration) {
+        Validate.notNull(configuration, "Video configuration is null!");
+        this.configuration = configuration;
+        this.frameRate = configuration.getFrameRate();
+        screenBounds = Toolkit.getDefaultToolkit().getScreenSize();
+    }
+
+    public void startRecording(final File toFile) {
+        if (isRecording()) {
+            throw new IllegalStateException("Already recording");
+        }
+
+        recordedVideo = toFile;
+
+        try {
+            recordedVideo.createNewFile();
+        } catch (IOException ex) {
+            throw new RuntimeException("Unable to create file to which video will be saved: " + recordedVideo.getAbsolutePath());
+        }
+
+        running = true;
+
+        thread = new Thread(new Runnable() {
+            @Override
+            public void run() {
+
+                Webcam webcam = Webcam.getDefault();
+                webcam.setViewSize(screenBounds);
+                webcam.open();
+
+                try {
+                    SequenceEncoder enc = new SequenceEncoder(recordedVideo);
+                    timer = new Timer();
+                    timer.schedule(new TestTimeoutTask(), TimeUnit.SECONDS.toMillis(configuration.getTestTimeout()));
+
+                    while (running) {
+                        BufferedImage bgrScreen = webcam.getImage();
+                        enc.encodeImage(bgrScreen);
+
+                        try {
+                            Thread.sleep(500 / frameRate);
```

Comment:
Neither do I, since it should be 1000 but since it was done in Desktop  I thought that maybe you found some kind of better pace doing in this way.